templates/template_data.go

```diff
@@ -57,6 +65,44 @@ type ExtendedData struct {
 	ExternalURL string `json:"externalURL"`
 }
 
+// FromContent calls Parse on all provided template content and returns the resulting Template. Content equivalent to templates.FromGlobs.
+func FromContent(tmpls []string, options ...template.Option) (*Template, error) {
+	t, err := New(options...)
+	if err != nil {
+		return nil, err
+	}
+
+	// Parse default template strings defined as files. Copied from template.FromGlobs.
+	// TODO: Consider removing this and defining them as strings. This would be a breaking change for those who defined custom default files.
+	defaultFileBasedTemplates := []string{"default.tmpl", "email.tmpl"}
```

Comment:
I most likely miss something but the code below reads from hardcoded (in-memory) assets, not hard disk. `/templates/default.tmpl` is actually https://github.com/grafana/prometheus-alertmanager/blob/4f0bdf4940594319336a1faa1126c4f4e11767f8/asset/assets_vfsdata.go#L163-L168

Reply:
Therefore, if I read the code correctly, the "for" clause below can be deleted if we do not use the default templates at all, which I think is true in Grafana Managed Alerts but not in Mimir, and therefore, it can be considered as breaking change for unification.

Reply:
Ah, no you didn't miss anything, got mixed up.  I'll remove the TODO entirely as I was confusing it with something else.